### PR TITLE
volontarily run dht in a go routine 

### PR DESCRIPTION
**Pull request by rkjdid**
following #44, I also applied the modification to the tests & examples, and included the minifix I mentionned about closing the d.portRequest chan

### CODE REVIEW

**Comment by rkjdid** [comment]
note also that it allows to remove the mechanism that uses the portRequest chan, since we know the port right after starting the dht

File: dht.go [hunk 1]
```diff
@@ -284,20 +284,40 @@ func (d *DHT) findNode(id string) {
...
 
-// Run starts a DHT node. It bootstraps a routing table, if necessary, and
-// listens for incoming DHT requests.
+func (d *DHT) Start() (err error) {
+	if err = d.initSocket(); err == nil {
+		go d.loop()
+	}
+	return err
+}
+
```
Comment (nictuku) [inline dht.go]: Add back the documentation for Run?

**Comment by nictuku** [comment]
Could you document Run and Start, explain their differences and indicate which one is deprecated?

File: dht_test.go [hunk 2]
```diff
@@ -32,7 +32,11 @@ func ExampleDHT() {
 		fmt.Println(err)
 		return
 	}
-	go d.Run()
+	err = d.Start()
```
Comment (nictuku) [inline dht_test.go]: minor: could you inline if err checking here and in the places you're handling errors below?

if err = d.Start(); err != nil {

**Comment by rkjdid** [comment]
of course, documentation!
do you want me to also add a depreciation `log.Warning` ?

**Comment by nictuku** [comment]
That would be nice, yep.

File: dht.go [hunk 3]
```diff
@@ -284,20 +284,57 @@ func (d *DHT) findNode(id string) {
...
+// Run launches the dht node. It starts a listener
+// on the desired address, then runs the main loop in the
+// same go routine.
+// If initSocket fails, Run returns with the error.
+// If initSocket succeeds, Run blocks until d.Stop() is called.
+// DEPRECIATED - Start should be used instead of Run
```
Comment (nictuku) [inline dht.go]: I think you mean DEPRECATED: http://stackoverflow.com/questions/9208091/the-difference-between-deprecated-depreciated-and-obsolete

**Comment by rkjdid** [comment]
Thanks for the heads up, same word in french for both ideas

**Comment by rkjdid** [comment]
closing this, let's squash it

**Pull request closed without merging**